sdk/src/main/scala/com/horizen/api/http/SidechainNodeApiRoute.scala

```diff
@@ -114,6 +116,26 @@ case class SidechainNodeApiRoute(peerManager: ActorRef,
     }
   }
 
+  def stop: Route = (post & path("stop")) {
+    try {
+      // we are stopping the node, and since we will shortly be closing network services lets do in a separate thread
+      // and give some time to the HTTP reply to be transmitted immediately in this thread
+      new Thread( new Runnable() {
+        override def run(): Unit = {
+          log.info("Sleeping 50 msec...")
+          sleep(50)
```

Comment:
50 seconds can be not enough on weak PCs. like on our CI. So can lead to random failures when of API.
I'm curious to know if AKKA http has something like "onComplete" event that can be used for a particular request. If yes we may try to use it and then execute `stopAll`.
Let's check it.

Reply:
I found no alternative, for the time being the number of millisec has been increased to 500